src/components/Modals/OptionsModal.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect, useRef } from "react";
+import {
+  DotsHorizontalIcon,
+  PencilIcon,
+  TrashIcon,
+} from "@heroicons/react/outline";
+
+export const OptionsModal = () => {
+  const [showOptionsModal, setShowOptionsModal] = useState(false);
+  const ref = useRef();
+  useEffect(() => {
+    const checkIfClickedOutside = e => {
+      if (showOptionsModal && ref.current && !ref.current.contains(e.target)) {
+        setShowOptionsModal(false);
+      }
+    };
+    document.addEventListener("mousedown", checkIfClickedOutside);
+
+    return () => {
+      document.removeEventListener("mousedown", checkIfClickedOutside);
+    };
+  }, [showOptionsModal]);
```

Comment:
You can extract this effect in hook because you might be using it in many places.